### DISCUSSION TITLE
Going automatically to the next row

### DISCUSSION THREAD

**Discussion by Zennro-hoa** [helgoboss/helgobox#1092] [Ideas]
Hi there! I'm new using playtime and I would like to know how/where is the option to set the next row when the previous has been cued ?
Wish you a nice evening !

**Comment by helgoboss** [comment]
Can you describe in more detail what you are looking for?

**Reply by Zennro-hoa** [reply]
Hi, thanks for your help!

This option exists for example in Ableton Live and is named Scene Follow Action. This person explains it at the beginning of this video : https://www.youtube.com/watch?v=ALw4UqN9u0s

Best,

**Reply by helgoboss** [reply]
Please request a new feature by creating a discussion in the Ideas section. There's a discussion for "Clip follow actions" already, but "Scene follow actions" are a different thing.

Or wait, I think you can just change the category of this discussion itself to "Ideas".

Thanks!

**Comment by antti-nykyri** [comment]
Hi, I would like very much to see this functionality as well. But in order to work for triggering complex setups of clips (and stopping clips at certain moments) for live events, I believe the Playtime matrix would need to also support optional stop buttons for slots as well. So that when you advance through your "cue list" or rows by just pressing one go button repeatedly to trigger your scenes one after another, all earlier clips wouldn't just stop but continue to play depending on if there's a stop button or a new clip on the track. I was a bit surprised to find out this fuctionality isn't included as it doesn't seem so complicated compared to everything included already, but I guess it might not be so important for just music making. The reason many sound designers/composers use Ableton Live for live events is this functionality, it's very fast and easy to setup like this and to edit your set during rehearsals. When you need to trigger a lot of clips (and stops) in a certain order during performance and to visually monitor stage events, triggering and stopping individual clips from a matrix just don't work. I would happily switch from Live to Reaper for performing live events, but this functionality would need to be included first. But thanks for the great software already, I bought Playtime to support the development and I'm interested to see how it will grow :)

**Reply by helgoboss** [reply]
For the sake of having everything at one place, here's what you wrote in the REAPER forum in the meanwhile:

> I've been following how Playtime gets developed with a lot of interest since the version 1. I've also been Ableton Live user since version 1.5 so I was naturally interested in "Session View à la Ableton Live". I've had hopes of moving all of my work including running live shows to Reaper solely, but there's something preventing that.
> 
> For performing arts Live is often used for launching audio clips, perhaps as a more musician oriented way of working opposed to the more strict "cue list" approach, which softwares such as Qlab use. Nevertheless it's possible to use Live for triggering audio and midi in a "cue list" kind of way, to launch all kind of combinations of events to play audio or even trigger lights or whatever devices with midi clips, if you need to. I've ran even outboard mixer automation with midi clips along with my audio clips, which worked well. When designing live performances as a sound designer, you can often end up with a large collection of things you have to launch by using scenes, scene by scene, while at the same time following the performance, playing with effect parameter control and even mixing the whole live sound of the show (band, live mics etc.). In my experience, two things in Live Session view enable it in this kind of scenarios:
> 
> 1) Being able to automatically advance the cursor to next scene, after one is launced. While this is not a necessity, its pretty important for busy shows in order to not screw up.
> 
> 2) Being able to assign stop buttons to slots or optionally not having those in slots. So that all of your playing audio and midi will not stop when you launch a new scene having information, which is supposed to play simultaneously with your previously triggered things.
> 
> I left a note regarding to this at Helgoboss Github page, but I'm unsure if that will trigger much conversation really. Since these features seem pretty lightweight (well for a person who knows nothing about programming, at least..) compared to some that Playtime already has, I wanted to ask if this is a direction you have considered? I would imagine adding these would allow a lot of new use cases for the software. As a Playtime 2 license owner I'm of course happy to test and comment anything regarding this, if something gets developed.
> 
> Anyway, thank You for a lot of great software development

**Reply by helgoboss** [reply]
Fundamentally, I'm not opposed to having a scene-follow action. At the point when I tackle clip-follow actions, I can also have a look at this.

However, I'd be interested in knowing more about this "Stop" button thing. I'm not very familiar with Live but the feature to assign stop buttons to slots always confused me. I thought it's a kind of cheap workaround to tackle a problem that could be much better solved in other ways. But maybe I'm wrong and I just didn't understand the real use case. Could you point me to a video where I can see such stop buttons in action? I would have a look at that video and check if it would make sense for Playtime as well. Maybe some things can be done better in Playtime (because of some fundamental new features that it adds to the mix, such as multi-slots and meta-clips) ... or maybe not. I'll see.

**Reply by antti-nykyri** [reply]
Thank you for replying!

That sounds great, looking forward to see perhaps the scene follow and also what kind of clip-follow actions there will be.

I can also try to put up a video that could demonstrate what I described, if you want me to. But lets see if I can describe an use case, so you see what I'm after:

When you create compositions and sound design for performing arts, there can often be things happening on stage (or wherever you are performing in) that have all kind of inner relations regarding to timings, order and how they progress. Usually (not always) you know the order of the things that are going to happen, but the exact timings and the way things progress simultaneously are not always fixed.

I have been creating ambientish music and sound design for contemporary dance a lot for example. It's quite common for me to have a live set, where the beginnings of the compositional parts of sound or music that I trigger are rehearsed together with dancers and we take great care of who gives the impulse for something to start (in music and coreography/dance) and who follows. But while a compositional part of the whole piece (including that piece of music and also coreography/dance) is taking place, there can also be simultaneous other things (music and dance) happening in the performance which have no strict or fixed relation regarding timing to this first part. So while they are ment to be played/danced/seen/heard at the sime time, they do not share a joint absolute "timeline" with other simultaneous things in this performance, which they would both try to exactly follow like in a singular musical piece. They rather co-exist and their relation to time vary inevitably. Often a varying length of certain (live) physical activity on stage can define the cue for launching more audio clips, so they will inevitably have no fixed relation to the audio that is already playing. So it's actually the musical composition that has to allow this flexibility, as the whole performance is not trying to fit and adapt into one absolute timing or singular "rhythm" of a pre-defined musical composition. I've actually published a research article about this concerning the absolute time of a recording and relative timings of a (live) performance, but it's unfortunately in Finnish.. Anyway, when you are working with ambient music and sound, creating this kind of flexibility for the composition is usually not a problem. In order to perform it you just need to be able to trigger all kind of audio and files with varying timings and let many of them to progress freely to their end while still being able to trigger all kind of new things with free timings at the same time.

In order to let certain clips to play until their end regardless of the following scene launches in Playtime, you would need to be able to disable the stop functionality from the slots that follow these clips in that column, right? There are no visible stop buttons now in Playtime, but every slot acts as it has one. You can't launch a scene wihout interfering with the clips that are already playing as you don't have the option to disable "stop" feature for selected empty slots. There could be lot's of ways to solve this and perhaps it wouldn't need to be a stop button. Anything that would disable the "stop" feature from triggering for a certain empty slot when the scene it belongs to is launched, would work.

The case I described might be a bit generalized, but I hope it demonstrates why being able to let previous clips to continue playing freely (or select to not let them play freely and to stop them, if that is the intention) while launching scenes, would make a big difference. In practice you might have a problem like: "How can I play a three minutes long drone sound from start to end on one column while simultaneously launcing rapidly many clips using scenes from other columns and timing the launch of these scenes to physical action on the stage, not to music?

Cheers!

**Reply by antti-nykyri** [reply]
For visual reference, here is a screenshot of a Ableton live set that was created and used tor a real show. On the left you can see the midi clips used for mixer mute automation and on the right side the audio clips and some dummy automation clips which are used for triggering volume automation for clips that were launched earlier from other tracks. Using dummy clips to trigger all kind of automation for other tracks has been a valuable feature btw, as you sometimes have to do automation that is separate from the audio itself, for tracks that have a clip already playing.

For this show I had to concentrate mostly on live mixing and I was triggering scenes in Ableton Live sequentially using a single button in my mixer, that sent just the scene launch command to Live. So I was just quickly checking that I am on right cue (scene) on Live and pressing the button on the right time repeatedly while mixing the show. Notice how most of the slots following audio clips on the same track have no stop buttons at all as most clips are ment to play until their end, independently from when the next scene is launched. Please also take notice of how there can be stop buttons on certain tracks (colums in Playtime language) for certain scenes, as those scenes are specifically there for stopping selected audio clips (and not others).

I hope this gives some reference on why the scene follow and selectable "stop/no stop" function for every independent slot would be invaluable for running live events based on using sequential scene launching (go-button type use)?

Cheers!

![Screenshot 2025-04-05 114723](https://github.com/user-attachments/assets/d7820e05-df16-49bb-a515-0bb843e7ffee)

**Comment by Zennro-hoa** [comment]
Hi Benjamin,

I couldn't find a video on internet so I did a short example for you in
Ableton. Is it clear to you ? I can also show other things for stop
buttons.

Best,
R

 Stop example_Ableton_720p.mov
<https://drive.google.com/file/d/16V_vDGqIj2E672dgRS_yH52i3rT7QDUI/view?usp=drive_web>


Le jeu. 3 avr. 2025 à 10:16, Benjamin Klum ***@***.***> a
écrit :

> Fundamentally, I'm not opposed to having a scene-follow action. At the
> point when I tackle clip-follow actions, I can also have a look at this.
>
> However, I'd be interested in knowing more about this "Stop" button thing.
> I'm not very familiar with Live but the feature to assign stop buttons to
> slots always confused me. I thought it's a kind of cheap workaround to
> tackle a problem that could be much better solved in other ways. But maybe
> I'm wrong and I just didn't understand the real use case. Could you point
> me to a video where I can see such stop buttons in action? I would have a
> look at that video and check if it would make sense for Playtime as well.
> Maybe some things can be done better in Playtime (because of some
> fundamental new features that it adds to the mix, such as multi-slots and
> meta-clips) ... or maybe not. I'll see.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/helgoboss/helgobox/discussions/1092#discussioncomment-12710814>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKPSCZYL4E7NJOQT4655TWL2XTU43AVCNFSM6AAAAABMMUXEBOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENZRGA4DCNA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Rozenn Lièvre

sie/ihr, she/her, elle
--------------------------------------------
*Sound Design*
*Sound Engineer*
--------------------------------------------
E-mail: ***@***.***
Tel (DE) : (+49) 0 176 58613088
Website: www.rozennlievre.com
---------------------------------------------

**Reply by helgoboss** [reply]
Thanks. Could you describe a bit what you do there and what's different from just using the column stop button?

**Comment by Zennro-hoa** [comment]
Hi Benjamin,

Here is a new video with sound.
Let me know what you think.

Best,
R

 Ableton_StopButtons.mp4
<https://drive.google.com/file/d/1tC0zR1iNtEFs-PpWNKmp-WNpMnakJDo_/view?usp=drive_web>



Le jeu. 3 avr. 2025 à 12:50, MegaUnbalanced ***@***.***> a
écrit :

> Thank you for replying!
>
> That sounds great, looking forward to see perhaps the scene follow and
> also what kind of clip-follow actions there will be.
>
> I can also try to put up a video that could demonstrate what I described,
> if you want me to. But lets see if I can describe an use case, so you see
> what I'm after:
>
> When you create compositions and sound design for performing arts, there
> can often be things happening on stage (or wherever you are performing in)
> that have all kind of inner relations regarding to timings, order and how
> they progress. Usually (not always) you know the order of the things that
> are going to happen, but the exact timings and the way things progress
> simultaneously are not always fixed.
>
> I have been creating ambientish music and sound design for contemporary
> dance a lot for example. It's quite common for me to have a live set, where
> the beginnings of the compositional parts of sound or music that I trigger
> are rehearsed together with dancers and we take great care of who gives the
> impulse for something to start (in music and coreography/dance) and who
> follows. But while a compositional part of the whole piece (including that
> piece of music and also coreography/dance) is taking place, there can also
> be simultaneous other things (music and dance) happening in the performance
> which have no strict or fixed relation regarding timing to this first part.
> So while they are ment to be played/danced/seen/heard at the sime time,
> they do not share a joint absolute "timeline" with other simultaneous
> things in this performance, which they would both try to exactly follow
> like in a singular musical piece. They rather co-exist and their relation
> to time vary inevitably. Often a varying length of certain (live) physical
> activity on stage can define the cue for launching more audio clips, so
> they will inevitably have no fixed relation to the audio that is already
> playing. So it's actually the musical composition that has to allow this
> flexibility, as the whole performance is not trying to fit and adapt into
> one absolute timing or singular "rhythm" of a pre-defined musical
> composition. I've actually published a research article about this
> concerning the absolute time of a recording and relative timings of a
> (live) performance, but it's unfortunately in Finnish.. Anyway, when you
> are working with ambient music and sound, creating this kind of flexibility
> for the composition is usually not a problem. In order to perform it you
> just need to be able to trigger all kind of audio and files with varying
> timings and let many of them to progress freely to their end while still
> being able to trigger all kind of new things with free timings at the same
> time.
>
> In order to let certain clips to play until their end regardless of the
> following scene launches in Playtime, you would need to be able to disable
> the stop functionality from the slots that follow these clips in that
> column, right? There are no visible stop buttons now in Playtime, but every
> slot acts as it has one. You can't launch a scene wihout interfering with
> the clips that are already playing as you don't have the option to disable
> "stop" feature for selected empty slots. There could be lot's of ways to
> solve this and perhaps it wouldn't need to be a stop button. Anything that
> would disable the "stop" feature from triggering for a certain empty slot
> when the scene it belongs to is launched, would work.
>
> The case I described might be a bit generalized, but I hope it
> demonstrates why being able to let previous clips to continue playing
> freely (or select to not let them play freely and to stop them, if that is
> the intention) while launching scenes, would make a big difference. In
> practice you might have a problem like: "How can I play a three minutes
> long drone sound from start to end on one column while simultaneously
> launcing rapidly many clips using scenes from other columns and timing the
> launch of these scenes to physical action on the stage, not to music?
>
> Cheers!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/helgoboss/helgobox/discussions/1092#discussioncomment-12712431>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKPSCZ6YJ3DLSUYDR5IWJRD2XUG6PAVCNFSM6AAAAABMMUXEBOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENZRGI2DGMI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Rozenn Lièvre

sie/ihr, she/her, elle
--------------------------------------------
*Sound Design*
*Sound Engineer*
--------------------------------------------
E-mail: ***@***.***
Tel (DE) : (+49) 0 176 58613088
Website: www.rozennlievre.com
---------------------------------------------

**Reply by helgoboss** [reply]
Hi @Zennro-hoa! Thanks a lot for the detailed explanations! I had a quick look already but will have to go through it again to understand all of your points.

Some things you showed, remind me a bit of existing features. Not sure if you know about them:

- Double clicking a stop button always ends the clip abruptly (without waiting for the quantization) - but that doesn't work with scenes because scenes can't be stopped.
- It's possible to exclude complete columns from being affected by disabling the scene-following button on the top of the column. But it can only be done for the whole column at once, not per slot.

I know, that's not the same as in Live. I will give it some thoughts.

**Reply by helgoboss** [reply]
I went through the video in detail and tried to identify what functionality already exists in Playtime and what is new. Please let me know about each point if I understood correctly. BTW, it's best if you view this on GitHub directly.

#### 1. Everything until 01:46 is essentially the same in Playtime (no new feature)

#### 2. At 01:46, you show the possibility to delete the stop button

The effect of it is that the scene play button doesn't stop clip "God Fear & Money" anymore.

Here I have the first question: Playtime has the option to "delete the stop button" for the complete column:
![image](https://github.com/user-attachments/assets/f98adbc7-ba18-4781-95cd-f43bb62a9714)

If I understand you correctly, that this is not enough because in a live show, you sometimes want to stop the other clips in the same column and sometimes not. Right?

#### At 03:00, you mention clip follow actions

This is already requested in #1044.

#### At 03:27, you mention scene follow actions

This is basically the current feature request (#1092).

#### At 04:36, you show the possibility to press enter to play the scene **and go down**

In Playtime, you can press Enter to play the scene, but it doesn't automatically go down to the scene below. I guess this is another FR?

**Comment by Zennro-hoa** [comment]
👍

Le ven. 4 avr. 2025 à 01:34, Benjamin Klum ***@***.***> a
écrit :

> Hi @Zennro-hoa <https://github.com/Zennro-hoa>! Thanks a lot for the
> detailed explanations! I had a quick look already but will have to go
> through it again to understand all of your points.
>
> Some things you showed, remind me a bit of existing features. Not sure if
> you know about them:
>
>    - Double clicking a stop button always ends the clip abruptly (without
>    waiting for the quantization) - but that doesn't work with scenes because
>    scenes can't be stopped.
>    - It's possible to exclude complete columns from being affected by
>    disabling the scene-following button on the top of the column. But it can
>    only be done for the whole column at once, not per slot.
>
> I know, that's not the same as in Live. I will give it some thoughts.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/helgoboss/helgobox/discussions/1092#discussioncomment-12719603>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKPSCZYDVXLXSWOW35GHQUL2XXAQRAVCNFSM6AAAAABMMUXEBOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENZRHE3DAMY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Rozenn Lièvre

sie/ihr, she/her, elle
--------------------------------------------
*Sound Design*
*Sound Engineer*
--------------------------------------------
E-mail: ***@***.***
Tel (DE) : (+49) 0 176 58613088
Website: www.rozennlievre.com
---------------------------------------------

**Comment by Zennro-hoa** [comment]
Hi Benjamin,

thanks for your reply.
Here is the answers:

If I understand you correctly, that this is not enough because in a live
show, you sometimes want to stop the other clips in the same column and
sometimes not. Right?
Correct.

In Playtime, you can press Enter to play the scene, but it doesn't
automatically go down to the scene below. I guess this is another FR?
Exactly.

Also very interesting what Mega Unbalanced said in his previous message.
Now I have a question for you: How could be be implemented in Playtime to
have this "automations clips" ? I mean, no audio, but the possibility to
just have automations saved in the clip ?


Best,
R



Le sam. 5 avr. 2025 à 11:34, MegaUnbalanced ***@***.***> a
écrit :

> For visual reference, here is a screenshot of a Ableton live set that was
> created and used tor a real show. On the left you can see the midi clips
> used for mixer mute automation and on the right side the audio clips and
> some dummy automation clips which are used for triggering volume automation
> for clips that were launched earlier from other tracks. Using dummy clips
> to trigger all kind of automation for other tracks has been a valuable
> feature btw, as you sometimes have to do automation that is separate from
> the audio itself, for tracks that have a clip already playing.
>
> For this show I had to concentrate mostly on live mixing and I was
> triggering scenes in Ableton Live sequentially using a single button in my
> mixer, that sent just the scene launch command to Live. So I was just
> quickly checking that I am on right cue (scene) on Live and pressing the
> button on the right time repeatedly while mixing the show. Notice how most
> of the slots following audio clips on the same track have no stop buttons
> at all as most clips are ment to play until their end, independently from
> when the next scene is launched. Please also take notice of how there can
> be stop buttons on certain tracks (colums in Playtime language) for certain
> scenes, as those scenes are specifically there for stopping selected audio
> clips (and not others).
>
> I hope this gives some reference on why the scene follow and selectable
> "stop/no stop" function for every independent slot would be invaluable for
> running live events based on using sequential scene launching (go-button
> type use)?
>
> Cheers!
>
> Screenshot.2025-04-05.114723.png (view on web)
> <https://github.com/user-attachments/assets/d7820e05-df16-49bb-a515-0bb843e7ffee>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/helgoboss/helgobox/discussions/1092#discussioncomment-12733416>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKPSCZ2RNDX6INS5OSPN3V32X6PQVAVCNFSM6AAAAABMMUXEBOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENZTGM2DCNQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Rozenn Lièvre

sie/ihr, she/her, elle
--------------------------------------------
*Sound Design*
*Sound Engineer*
--------------------------------------------
E-mail: ***@***.***
Tel (DE) : (+49) 0 176 58613088
Website: www.rozennlievre.com
---------------------------------------------

**Comment by antti-nykyri** [comment]
> How could be be implemented in Playtime to
have this "automations clips" ? I mean, no audio, but the possibility to
just have automations saved in the clip ?

This indeed is something that in Live is possible by using dummy clips, but it's kind of an "hack" from user perspective. It goes round a limitation which Live has in order to do something in a bit cumbersome way. It's not really a feature of Live but more like a way of getting round the limitations in Live, something that users have found possible to do. If implementation of native automation (only) clips (including track/column parameters like volume, pan etc. and effect parameters assigned to that specific track) could be done in Playtime in a way that would not stop or affect the previously launced audio on that track/column, it would be something special compared to Live. Even having automation clips only for effect parameters in that column/track would help a lot, as you could always use volume faders, panner devices etc. as effects in track and automate those. Automating sends in automation clips would be useful though, but I believe most useful would be the ability to automate fx parameters on that track. In Live you have to mess with track routings, extra tracks etc. in order to launch automation for your audio as a separate item/event. If in Playtime you could visually see the automation clips on the same column with your audio clips, that would be great and simplify things a lot too.

Sorry for including yet another feature to this discussion. We can create a separate thread for this one if needed. I wanted to mention this as all these features are kind of linked together and dependent on each other.

**Comment by Zennro-hoa** [comment]
Hi, yes I use this hack also in Ableton. I was wondering how is it possible
to do it in Playtime.

Best,
R

Le lun. 7 avr. 2025 à 13:12, MegaUnbalanced ***@***.***> a
écrit :

> How could be be implemented in Playtime to
> have this "automations clips" ? I mean, no audio, but the possibility to
> just have automations saved in the clip ?
>
> This indeed is something that in Live is possible by using dummy clips,
> but it's kind of an "hack" from user perspective. It goes round a
> limitation which Live has in order to do something in a bit cumbersome way.
> It's not really a feature of Live but more like a way of getting round the
> limitations in Live, something that users have found possible to do. If
> implementation of native automation (only) clips (including track/column
> parameters like volume, pan etc. and effect parameters assigned to that
> specific track) could be done in Playtime in a way that would not stop or
> affect the previously launced audio on that track/column, it would be
> something special compared to Live. In Live you have to mess with track
> routings, extra tracks etc. in order to launch automation for your audio as
> a separate item/event. If in Playtime you could visually see your
> automation clips on the same column with your audio clips, that would be
> great and simplify things a lot.
>
> Sorry for including yet another feature for this discussion. We can create
> a separate thread for this one if needed. I wanted to mention this as all
> these features are kind of linked together and dependent on each other.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/helgoboss/helgobox/discussions/1092#discussioncomment-12749640>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKPSCZ667GBX3KEDUS7PWJT2YJMRRAVCNFSM6AAAAABMMUXEBOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENZUHE3DIMA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Rozenn Lièvre

sie/ihr, she/her, elle
--------------------------------------------
*Sound Design*
*Sound Engineer*
--------------------------------------------
E-mail: ***@***.***
Tel (DE) : (+49) 0 176 58613088
Website: www.rozennlievre.com
---------------------------------------------
